Lib/contextlib.py

```diff
@@ -128,6 +128,10 @@ def __exit__(self, type, value, traceback):
                 # fixes the impedance mismatch between the throw() protocol
                 # and the __exit__() protocol.
                 #
+                # This cannot use 'except BaseException as exc' (as in the
+                # async implementation) to maintain compatibility with
+                # Python 2, where string exceptions are not caught by
+                # 'except BaseException'.
```

Comment:
The problem in 2.x is old style classes rather than strings:

```
>>> class LegacyException:
...     pass
... 
>>> raise LegacyException
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
__main__.LegacyException: <__main__.LegacyException instance at 0x7f1208568c68>
```